File: Managerial/DAL/Models/Batch.cs
using DAL.Core.Helpers;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System.Collections.Generic;

namespace DAL.Models
{
    public class Batch : AuditableEntity
    {
        public int? ProductId { get; set; }
        public int? InventoryId { get; set; }
        public int? StockId { get; set; }
        public int? WarehouseItemId { get; set; }
        public virtual Product Product { get; set; }
        public virtual Inventory Inventory { get; set; }
        public virtual Stock Stock { get; set; }
        public virtual ICollection<WarehouseItem> WarehouseItems { get; set; } = new HashSet<WarehouseItem>();
    }

    public class BatchConfig : IEntityTypeConfiguration<Batch>
    {
        public void Configure(EntityTypeBuilder<Batch> builder)
        {
            builder.BaseEntityBuilder();
        }
    }
}

File: Managerial/Managerial/ClientApp/src/app/models/product/Batch.ts
import { BaseModel } from "../BaseModel.model"
import { Inventory } from "../inventory/Inventory.model"
import { Stock } from "../inventory/stock.model"
import { WarehouseItem } from "../warehouse/WarehouseItem.model"
import { Product } from "./Product.model"

export class Batch extends BaseModel {
       public Product: Product
        public Inventory: Inventory
        public Stock: Stock
        public WarehouseItems: WarehouseItem
}
